src/RefractParseResult.cc

```diff
@@ -0,0 +1,88 @@
+//
+//  RefractParseResult.cc
+//  drafter
+//
+//  Created by Pavan Kumar Sunkara on 25/09/15.
+//  Copyright (c) 2015 Apiary. All rights reserved.
+//
+
+#include "RefractAPI.h"
+#include "RefractParseResult.h"
+
+namespace drafter {
+
+    typedef std::vector<refract::IElement*> RefractElements;
+
+    refract::IElement* SourceMapRowToRefract(const mdp::CharactersRange& sourceMap)
+    {
+        refract::ArrayElement* element = new refract::ArrayElement;
+
+        element->push_back(refract::IElement::Create((int) sourceMap.location));
+        element->push_back(refract::IElement::Create((int) sourceMap.length));
+
+        return element;
+    }
+
+    refract::IElement* SourceMapToRefract(const mdp::CharactersRangeSet& sourceMap)
+    {
+        refract::ArrayElement* element = new refract::ArrayElement;
+        RefractElements content;
+
+        element->element(SerializeKey::SourceMap);
+
+        for (mdp::CharactersRangeSet::const_iterator it = sourceMap.begin();
```

Comment:
Please use `std::transform` instead of `for` loop.
```
std::transform(sourceMap.begin(), sourceMap.end(), std::back_inserter(content), SourceMapRowToRefract);
```
It is easier to understand to one line of code than to six lines.